gufe/protocols/protocol.py

```diff
@@ -192,9 +193,9 @@ def create(
             The starting `ChemicalSystem` for the transformation.
         stateB : ChemicalSystem
             The ending `ChemicalSystem` for the transformation.
-        mapping : Optional[dict[str, ComponentMapping]]
+        mapping : Optional[ComponentMapping | list[ComponentMapping]]
```

Comment:
```suggestion
        mapping : Optional[ComponentMapping, list[ComponentMapping]]
```

As per the Python docs Optional is a special type of Union, Union[A, B] is A | B.
I.e. Union[A | B] would be the same as saying Union[Union[A, B]] rather than Union[A, B]

Reply:
that syntax is 3.10 only, I think we're still supporting 3.9 per nep29

Reply:
@richardjgowers https://docs.python.org/3.9/library/typing.html#typing.Union 

That's the correct syntax for py3.9?